quality/integration/src/test/java/test/integration/auth/RoleAdminTaskTest.java

```diff
@@ -225,4 +245,31 @@ public void testCheckRole()
             assertEquals(actual.subList(1, actual.size()), entry.getValue());
         }
     }
+
+    @Test
+    public void testCreateRoleWithUnboundPermissions() throws Exception  {
+        // First, give a new role all permissions within the "sor" context
+        _roleManager.createRole(new RoleIdentifier(null, "sor-only"), new RoleUpdateRequest()
+                .withPermissionUpdate(new PermissionUpdateRequest().permit(
+                        ImmutableSet.of("sor|*"))));
+
+        _authIdentityManager.updateIdentity(new ApiKey("sor-only-key", "id5", ImmutableSet.of("sor-only")));
+
+        StringWriter out = new StringWriter();
+
+        _task.execute(ImmutableMultimap.of(
+                ApiKeyRequest.AUTHENTICATION_PARAM, "sor-only-key",
+                "action", "update",
+                "role", "new-role",
+                "permit", "sor|update|if({..,\"foo\":\"bar\"})",
+                "permit", "queue|post|*"),
```

Comment:
I agree that it opens the door for redundant roles, but I don't think it should be prohibited.  For example, it may be that the new role _today_ shares the same permissions, but by creating the new role it can be modified in the future without needing to be concerned about either unintended consequences of modifying the existing role or hunting down all users with the reused role and re-assigning their roles in the future.